### PR TITLE
feat: Add GKE clusters

### LINKED ISSUE #255: GKE clusters resource
### Describe the resource.

Ingest GKE clusters resource to include clusters, specifically the node count at any given time associated with them. Could also poll for further details on node pools

### Use Case

Ability to track total count of nodes per cluster in GKE for comparison to count of known clusters/nodes seen by security tools.

### Additional context

Relevant APIs are documented in https://cloud.google.com/kubernetes-engine/docs/reference/rest

Node count of a cluster is returned as part of the projects.locations.clusters.list operation results  currentNodeCount

https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Fixes https://github.com/cloudquery/cq-provider-gcp/issues/255
~~Blocked by https://github.com/cloudquery/cq-provider-sdk/pull/268 and https://github.com/cloudquery/cq-provider-gcp/pull/268~~

We can remove the migration files once https://github.com/cloudquery/cq-provider-gcp/pull/271 is merged

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run --new-from-rev main` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

**Comment by erezrokah** [comment]
I have a guess that somehow the compute and container resources are conflicting hence the tests are failing on this PR.

**Comment by erezrokah** [comment]
Blocked by https://github.com/cloudquery/cq-provider-gcp/pull/275

File: resources/provider/provider.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ func Provider() *provider.Provider {
 			"sql.instances":                sql.SQLInstances(),
 			"domains.registrations":        domains.DomainsRegistration(),
 			"bigquery.datasets":            bigquery.BigqueryDatasets(),
+			"container.clusters":           container.Clusters(),
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply resources/provider/provider.go:84]: Ah, I followed the naming convention of the GCP golang API.
Definitely can use the console term of `kubernetes`.

[the remark this answers is not in the document]
Comment (erezrokah) [reply resources/provider/provider.go]: Fixed in https://github.com/cloudquery/cq-provider-gcp/pull/267/commits/12a26db89d153f87e1b5a8ac3b49208baf53f1d0

**Comment by erezrokah** [comment]
Testing are failing due to https://github.com/cloudquery/cq-provider-sdk/pull/283

File: resources/services/kubernetes/clusters.go [hunk 2]
```diff
@@ -0,0 +1,1185 @@
...
+
+	res <- output.Clusters
+	return nil
+}
+
+func castParentToCluster(parent *schema.Resource) (*container.Cluster, error) {
+	p, ok := parent.Item.(*container.Cluster)
+	if !ok {
+		return nil, fmt.Errorf("expected *container.Cluster but got %T", p)
+	}
```
Comment (bbernays) [inline resources/services/kubernetes/clusters.go:1115]: I don't think we are supposed to be checking that the cast was successful anymore

**Comment by erezrokah** [comment]
Looking at the integration tests failures as the pass locally for me on our dev account, but not on our prod account

File: terraform/kubernetes/prod/terraform.tf [hunk 3]
```diff
@@ -1,6 +1,6 @@
 terraform {
   backend "gcs" {
     bucket = "cq-provider-gcp-tf-state"
-    prefix = "container"
+    prefix = "kubernetes"
```
Comment (erezrokah) [inline terraform/kubernetes/prod/terraform.tf:4]: I manually cleaned up the old state of `container` due to this rename

**Review by bbernays — COMMENTED** [review]
LGTM, just added a few other people to make sure we get a thorough review

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**